### PR TITLE
Added note that mixed-version cluster states cause cluster restore fail

### PR DESCRIPTION
**Pull request by kathancox**
Fixes DOC-5503

Add a reference that when a cluster is upgrading and in a mixed-version state, a full-cluster restore will fail.

### CODE REVIEW

File: v22.2/restore.md [hunk 1]
```diff
@@ -159,12 +159,13 @@ Also, a full cluster restore will:
 
 - Restore [temporary tables](temporary-tables.html) to their original database during a full cluster restore.
 - Drop the cluster's `defaultdb` and `postgres` [pre-loaded databases](show-databases.html#preloaded-databases) before the restore begins. You can only restore `defaultdb` and `postgres` if they are present in the original [backup](take-full-and-incremental-backups.html).
+- When the cluster is in a mixed-version state during an [upgrade](upgrade-cockroach-version.html), a full cluster restore will fail.
```
Comment (kathancox) [inline v22.2/restore.md:162]: Hey @stevendanna I gleaned from the PR that full cluster restores **will** fail, but is that true?

**Comment by kathancox** [comment]
Hey @stevendanna, just a friendly review ping on this!

File: v22.2/restore.md (hunk 1 above)
Comment (stevendanna) [reply v22.2/restore.md:162]: Yup!  If you wanted to say more, you can say something like "To perform a full cluster restore, you can finalize the upgrade."

**Review by stevendanna — APPROVED** [review]

File: v22.2/restore.md [hunk 2]
```diff
@@ -159,12 +159,13 @@ Also, a full cluster restore will:
 
 - Restore [temporary tables](temporary-tables.html) to their original database during a full cluster restore.
 - Drop the cluster's `defaultdb` and `postgres` [pre-loaded databases](show-databases.html#preloaded-databases) before the restore begins. You can only restore `defaultdb` and `postgres` if they are present in the original [backup](take-full-and-incremental-backups.html).
+- When the cluster is in a mixed-version state during an [upgrade](upgrade-cockroach-version.html), a full cluster restore will fail. To perform a full cluster restore, it is necessary to [finalize the upgrade](upgrade-cockroach-version.html#step-3-decide-how-the-upgrade-will-be-finalized).
```
Comment (andf-crl) [inline v22.2/restore.md:162]: nit: think we just need `first` in here somewhere, maybe:

> To perform a full cluster restore, it is necessary to first [finalize the upgrade].

or maybe:

> to [finalize the upgrade] beforehand.

Up to you!

**Review by andf-crl — APPROVED** [review]
LGTM, one optional nit.

File: v22.2/restore.md (hunk 2 above)
Comment (kathancox) [reply v22.2/restore.md]: Done

**Comment by kathancox** [comment]
TFTRs!

**Pull request merged**